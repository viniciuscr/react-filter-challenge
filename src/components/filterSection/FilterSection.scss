@import "../../colors.scss";

.bg-gray {
  background-color: $bright-gray;
}

.Collapsible {
  border-top: 1px black solid;
}

.Collapsible__contentInner > ul {
  margin: 0;
  padding: 0;
}

.Collapsible__contentInner li {
  margin-bottom: 0.5rem;
}

.Collapsible__contentInner input[type="checkbox"] {
  margin-left: 0.5rem;
}
.Collapsible__contentInner label {
  width: 92%;
  margin: 0;
  padding-left: 0.5rem;
  font-weight: 600;
}
.Collapsible__trigger div {
  cursor: pointer;
}

/* Collapsible already has a logic for when it is opened or closed.
  We can delegate to it the control over the chevron
*/
.is-closed .icon-rl-chevron:before {
  content: "\e91d";
}
.is-open .icon-rl-chevron:before {
  content: "\e91e";
}