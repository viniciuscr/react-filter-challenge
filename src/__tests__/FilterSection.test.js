import React from "react";
import { mount } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import FilterSection from "../components/filterSection";

configure({ adapter: new Adapter() });

describe("FilterSection tests", () => {
  it("should render as expected", () => {
    const wrapper = mount(
      <FilterSection
        title="Section Test"
        options={["option", "option two", "option three"]}
        filterFunc={jest.fn()}
      />
    );
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });

  it("should call filterFunc when a item is clicked", () => {
    const filterFunc = jest.fn();
    const wrapper = mount(
      <FilterSection
        title="Section"
        options={["option", "option two", "option three"]}
        filterFunc={filterFunc}
      />
    );

    const option = wrapper.find("Filter").find("#Section-option");
    option.simulate("change");

    expect(filterFunc).toHaveBeenCalled();

    wrapper.unmount();
  });
});
