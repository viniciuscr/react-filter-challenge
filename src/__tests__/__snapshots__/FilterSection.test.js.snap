// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSection tests should render as expected 1`] = `"<div class=\\"Collapsible\\"><span class=\\"Collapsible__trigger is-open\\"><div class=\\"d-flex justify-content-between mt-1 mb-1 mr-1 \\"><h6>Section Test</h6><span class=\\"icon-rl-chevron\\"></span></div></span><div class=\\"Collapsible__contentOuter\\" style=\\"height: auto; transition: none; overflow: hidden;\\"><div class=\\"Collapsible__contentInner\\"><ul><li><div class=\\"bg-gray p-1 text-dark  rounded\\"><input name=\\"option\\" type=\\"checkbox\\" id=\\"Section Test-option\\" value=\\"option\\"><label for=\\"Section Testoption\\">option</label></div></li><li><div class=\\"bg-gray p-1 text-dark  rounded\\"><input name=\\"option two\\" type=\\"checkbox\\" id=\\"Section Test-option two\\" value=\\"option two\\"><label for=\\"Section Testoption two\\">option two</label></div></li><li><div class=\\"bg-gray p-1 text-dark  rounded\\"><input name=\\"option three\\" type=\\"checkbox\\" id=\\"Section Test-option three\\" value=\\"option three\\"><label for=\\"Section Testoption three\\">option three</label></div></li></ul></div></div></div>"`;
